config/config.go

```diff
@@ -263,6 +263,34 @@ func (c *toEnable) LogConfig(logger *logrus.Entry) {
 	logger.Info("enabled")
 }
 
+type SourceLoadingConfig struct {
+	DownloaderConfig DownloaderConfig `yaml:",inline"`
+
+	Concurrency        uint `yaml:"concurrency" default:"4"`
+	MaxErrorsPerSource int  `yaml:"maxErrorsPerSource" default:"5"`
+}
+
+func (c *SourceLoadingConfig) LogConfig(logger *logrus.Entry) {
+	logger.Infof("concurrency = %d", c.Concurrency)
+	logger.Infof("maxErrorsPerSource = %d", c.MaxErrorsPerSource)
+}
+
+type DownloaderConfig struct {
+	Timeout  Duration `yaml:"timeout" default:"5s"`
+	Attempts uint     `yaml:"attempts" default:"3"`
+	Cooldown Duration `yaml:"cooldown" default:"500ms"`
```

Comment:
Should be documented in docs/configuration.md and docs/config.yml

Reply:
Yeah I had made minor changes to the files locally to not forget, but was already thinking of supporting `RefreshPeriod` and `StartStrategy` in the hosts file resolver so wanted to wait until after that to update everything.  
Will get to it soon, still need to bring back the old options for back-compat.